### DISCUSSION TITLE
Volume control via GPIO

### DISCUSSION THREAD

**Discussion by rav345** [pschatzmann/ESP32-A2DP#432] [Q&A] [answered]
Hi,
I would like to control the volume of the DAC output of the A2DP target device (ESP32) via a signal input to one of the GPIO pins. How do I achieve this using the ESP32-A2DP library? I need the volume to be reduced relative to the current volume. Also, is it possible to issue play/pause commands via the pins?
Thanks

**Comment by pschatzmann** [comment]
I don't understand your question: All the necessary API is available. Are you maybe just to lazy to look it up yourself ?
And of cause I can't give you basic Arduino support!

**Reply by rav345** [reply]
Lol. It is hard to digest when someone whose work you admire calls you lazy instead of stopping at I don't understand your question.

**Comment by pschatzmann** [comment, marked as answer]
I am sorry, but yesterday I had an operation and I am still in the hospital suffering quite badly. 

The readme tells you: "Please read the documentation and check the issues and discussions before posting any new ones on Github."  

If you click on the class documentation link you should easily find the set_volume method. There is also a separate chapter for the volume conrol in the wiki.

**Reply by rav345** [reply]
Thanks, Phil. 
Sorry to hear about your troubles. Please rest well and get well soon! 
Gute Besserung.
Warmly,
R

**Comment by clarkbremer** [comment]
Phil has put so much effort into the documentation, so can imagine the
frustration when people don't take the time to read it.

sennt frommy mobile phone with a tinny keybard

**Reply by rav345** [reply]
There is usually so much lost in communication. Add to that the rabbit hole we all go into during technical problem-solving and figuring out specific details, often under pressure. If things were as easy as posting a comment on Twitter, nobody would need to post any questions here :-) 
Everything is evident when you figure it out, isn't it?
